logstash-core/src/main/java/org/logstash/jackson/StreamReadConstraintsUtil.java

```diff
@@ -0,0 +1,134 @@
+            this.propertyName = PROP_PREFIX + this.name().toLowerCase().replace('_', '-');
+            this.applicator = applicator;
+            this.observer = observer;
+        }
+
+        @FunctionalInterface
+        interface IntValueObserver extends Function<StreamReadConstraints, Integer> {}
+
+        @FunctionalInterface
+        interface IntValueApplicator extends BiFunction<StreamReadConstraints.Builder, Integer, StreamReadConstraints.Builder> {}
+    }
+
+    /**
+     * @return an instance configured by {@code System.getProperties()}
+     */
+    public static StreamReadConstraintsUtil fromSystemProperties() {
+        return new StreamReadConstraintsUtil(System.getProperties());
+    }
+
+    StreamReadConstraintsUtil(final Properties properties) {
+        this(properties, null);
+    }
+
+    StreamReadConstraintsUtil(final Properties properties,
+                              final Logger logger) {
+        this(extractProperties(properties), logger);
+    }
+
+    static private Map<String,String> extractProperties(final Properties properties) {
+        return properties.stringPropertyNames().stream()
+                .filter(propName -> propName.startsWith(Override.PROP_PREFIX))
+                .map(propName -> Map.entry(propName, properties.getProperty(propName)))
+                .filter(entry -> entry.getValue() != null)
+                .collect(Collectors.toUnmodifiableMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private StreamReadConstraintsUtil(final Map<String,String> propertyOverrides,
+                              final Logger logger) {
+        this.propertyOverrides = Map.copyOf(propertyOverrides);
+        this.logger = Objects.requireNonNullElseGet(logger, () -> LogManager.getLogger(StreamReadConstraintsUtil.class));
+    }
+
+    StreamReadConstraints get() {
+        if (configuredStreamReadConstraints == null) {
+            final StreamReadConstraints.Builder builder = StreamReadConstraints.defaults().rebuild();
+
+            eachOverride((override, value) -> override.applicator.apply(builder, value));
+
+            this.configuredStreamReadConstraints = builder.build();
+        }
+        return configuredStreamReadConstraints;
+    }
+
+    public void applyAsGlobalDefault() {
+        StreamReadConstraints.overrideDefaultStreamReadConstraints(get());
+    }
+
+    public void validateIsGlobalDefault() {
+        validate(StreamReadConstraints.defaults());
+    }
+
+    private void validate(final StreamReadConstraints streamReadConstraints) {
+        final List<String> fatalIssues = new ArrayList<>();
+        eachOverride((override, specifiedValue) -> {
+            final Integer effectiveValue = override.observer.apply(streamReadConstraints);
+            if (Objects.equals(specifiedValue, effectiveValue)) {
+                logger.info("Jackson default value override `{}` configured to `{}`", override.propertyName, specifiedValue);
```

Comment:
Currently, this will log at `info` for the default Logstash Jackson config set in `config/jvm.options`: 

```
[2024-12-04T12:53:30,180][INFO ][org.logstash.jackson.StreamReadConstraintsUtil] Jackson default value override `logstash.jackson.stream-read-constraints.max-string-length` configured to `200000000`
[2024-12-04T12:53:30,180][INFO ][org.logstash.jackson.StreamReadConstraintsUtil] Jackson default value override `logstash.jackson.stream-read-constraints.max-number-length` configured to `10000`
```

What do you think about logging at `debug`, or verifying against the Logstash defaults?

Reply:
It replaces identical info-level logging from before:

https://github.com/elastic/logstash/pull/16719/files#diff-8779c222532118bcfe1d6ccd7f5152238150615ec0b0d394d44e4fbcf334f903L66

I hesitate to add functionality to compare it against logstash defaults, when those defaults are currently held in and provided by a user-overridable config file.